ingestion/functions/parsing/cuba/cuba.py

```diff
@@ -0,0 +1,180 @@
+    from one of these countries in last 30 days if populated.
+
+    Diagnostic centre and treatment hospital are both included in notes for now, could be geocoded in future.
+
+    Currently no parsing of symptoms, as field is always left empty - worth rechecking this in future in case this field
+    becomes populated. No disease outcome data is provided either.
+
+    """
+    with open(raw_data_file, "r") as f:
+        json_data = json.load(f)
+        # First make dict mapping code names of diagnostic and treament centers
+        # to actual locations
+        hospital_map = {}
+        for centre_type in ['centros_aislamiento', 'centros_diagnostico']:
+            for centre in json_data[centre_type]:
+                hospital_map[centre] = json_data[centre_type][centre]['nombre'] + \
+                    ", " + json_data[centre_type][centre]['provincia']
+
+        for day in json_data['casos']['dias']:
+            if 'diagnosticados' in json_data['casos']['dias'][day]:
+                fecha = json_data['casos']['dias'][day]['fecha']
+                for entry in json_data['casos']['dias'][day]['diagnosticados']:
+                    notes = []
+                    case = {
+                        "caseReference": {
+                            "sourceId": source_id,
+                            "sourceEntryId": entry["id"],
+                            "sourceUrl": source_url},
+                        "location": convert_location(entry),
+                        "events": [
+                            {
+                                "name": "confirmed",
+                                "dateRange": {
+                                    "start": convert_date(fecha),
+                                    "end": convert_date(fecha)}}],
+                        "demographics": {
+                            "ageRange": {
+                                "start": float(
+                                    entry["edad"]),
+                                "end": float(
+                                    entry["edad"])},
+                            "gender": convert_gender(
+                                entry["sexo"]),
+                            "nationalities": [
+                                convert_nationality(
+                                    entry['pais'])]}}
+
+                    if entry.get("consulta_medico", ""):
+                        try:
+                            case["events"].append({
+                                "name": "firstClinicalConsultation",
+                                "dateRange": {
+                                    "start": convert_date(entry["consulta_medico"]),
+                                    "end": convert_date(entry["consulta_medico"])
+                                }}
+                            )
+                        except BaseException:
```

Comment:
what are you trying to catch/avoid here?

Reply:
errors trying to convert date time - I've now moved try/catch to convert_date function